pulser-core/pulser/math/abstract_array.py

```diff
@@ -71,6 +71,11 @@ def is_tensor(self) -> bool:
         """Whether the stored array is a tensor."""
         return self.has_torch() and isinstance(self._array, torch.Tensor)
 
+    @property
+    def is_differentiable(self) -> bool:
```

Comment:
Wouldn't `requires_grad` or `has_grad` be a more generic name for this property ? I have the feeling it does not only apply to differentiability (differentiability requires_grad, but requires_grad can be used for other perspectives, right ?)

Reply:
I used `is_differentiable` because I felt it was more explicit. `requires_grad` is clear for people familiar with torch but I find it quite opaque otherwise. 
Also, to me being differentiable = having a gradient. What do you have in mind that would fall outside of this scope?

Reply:
`is_differentiable` is definitely explicit. My only point is what this function does is what a function `requires_grad`  should do. We could imagine that at some point, `is_differentiable` can be different that `requires_grad` (might be conditions on the gradient, idk), I would prefer having two properties, one `requires_grad` and one `is_differentiable`, with `is_differentiable` currently returning the `requires_grad` property (I think that some people will still try to check the differentiability using `requires_grad`).

But it's very minor ;)

Reply:
Tbh I think it's overkill to bloat the source code with two ways to access the same property, at least at this stage. If you feel like `requires_grad` is better, I'll replace it